grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java

```diff
@@ -223,15 +222,27 @@ private void sendQueriesWhenBatchLargerThanValue(int value) {
      */
     public void waitToFinish(){
         flush();
+
+        //TODO: Check if this active block is actually required
         while(!futures.values().stream().allMatch(CompletableFuture::isDone)
                 && blocker.availablePermits() != blockerSize){
             try {
                 Thread.sleep(500);
             } catch (InterruptedException e) {
-                LOG.error(e.getMessage());
+                printError(e.getMessage());
+            }
+        }
+
+        for (CompletableFuture completableFuture : futures.values()) {
+            try {
+                //TODO: Figure out why a null sometimes gets put in here
```

Comment:
haha what

Reply:
Ooooo wait I did figure it out! Need to remove that comment and check